mdsa/approximate_mds.py

```diff
@@ -161,6 +164,23 @@ def principal_coordinates_analysis(distance_matrix, algorithm, dimensions):
         coords = array(comb_mds.getFinalMDS())
         eigvals = ['NaN'] * dimensions
         pcnts = ['NaN'] * dimensions
+    elif algorithm == 'svd':
+        # Singular value decomposition. We can assume that distance_matrix is symmetric.
+        # full_matrices=False so U and Vh are of shape (M,K) and (K,N) where K=min(M,N)
+        U, s, Vh = linalg.svd(distance_matrix, full_matrices=False)
+
+        # eigenvectors (without the imaginary component)
+        eigvecs = np.array(U.real)
+
+        # for our output coordinates, only pick the desired dimensions
+        coords = eigvecs[:, :dimensions]
+
+        # eigenvalues without the imaginary component
+        eigvals = np.array(s.real)[:dimensions]
+
+        # percentages (how much variation each coord is explaining)
+        pcnts = ((eigvals / np.sum(eigvals)) * 100)[:dimensions]
```

Comment:
note to self: return full solution (do not reduce to k dimensions). 

Also raise a warning that says: dimensions parameter ignored for this method (python warnings module: warnings.warn)